org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/SourceAssistProcessor.java

```diff
@@ -138,8 +140,33 @@ public SourceAssistProcessor(PreferenceManager preferenceManager) {
 
 		// Generate hashCode() and equals()
 		if (supportsHashCodeEquals(context, type, monitor)) {
-			Optional<Either<Command, CodeAction>> hashCodeEquals = getHashCodeEqualsAction(params);
-			addSourceActionCommand($, params.getContext(), hashCodeEquals);
+			// Generate QuickAssist
+			Optional<Either<Command, CodeAction>> quickAssistHashCodeEquals = Optional.empty();
+			ASTNode node = context.getCoveredNode();
+			if (node == null) {
+				node = context.getCoveringNode();
+			}
+			if (node instanceof SimpleName) {
+				ASTNode parent = node.getParent();
+				if (parent instanceof TypeDeclaration) {
+					quickAssistHashCodeEquals = getHashCodeEqualsAction(params, JavaCodeActionKind.QUICK_ASSIST);

```

Comment:
This exposes "Generate Getter and Setter..." code action menu at bulb of type declaration, that looks good to me.

![image](https://user-images.githubusercontent.com/14052197/130185422-5cb97c2d-cd85-49fa-b09c-19877b3755e1.png)

Reply:
currently it appears only when i hover at the type name, but not in the other place of the same line.

but if you have a look at "Move", it also appears at the "public" modifier. I think that's better.

![image](https://user-images.githubusercontent.com/14052197/131461530-8d892874-c876-4d0f-97fc-b5a781a2d9f7.png)

Reply:
![Screenshot 2021-08-31 191007](https://user-images.githubusercontent.com/45906942/131492599-fdbc6791-3525-4bfc-8960-4be5b074b506.png)

In the above screenshot, we can get the following information, including it's **startpotision** and **length**:
- `@entity`, `@Table(name = "owners")` and `public`: modifiers
- `Owner`: Class itself
- `Person`: extended class
- `Serializable`: implemented interface

For extending the place to show the light bulb, I have the following two plans:

1. When the covered node or the covering node is one of the above nodes, we offer the light bulb (in the screenshot, the six red squares will be valid)
2. calculate the first **startpotision** and the last **endpotision** as construct a range. If the covered node or the covering node is in the range, we offer the light bulb (in the screenshot, all parts between line 46 - line 49 will be valid)

@testforstephen which one do you prefer?

Reply:
Use 1 to make it consistent with "Move...".